From 9c5b7f53af55f92ab225d3fd54e3a362e3e7201e Mon Sep 17 00:00:00 2001
From: Jacky Chou <jacky_chou@aspeedtech.com>
Date: Wed, 6 May 2026 11:06:04 +0800
Subject: [PATCH] dts: arm: ast2705: simplify XGMAC DT node

Remove MTL RX/TX queue config sub-nodes, pcs-handle property and the
snps,dwmac-mdio compatible from the MDIO child node. The MTL queue
setup is now handled by the driver defaults, and XPCS is managed via
the PHY framework.

Upstream-Status: Backport [SDK v11.03 migration]
Signed-off-by: Jacky Chou <jacky_chou@aspeedtech.com>
Change-Id: I12dcd83a0d10ddc4c182aff8a2d8a61142c1e521
Signed-off-by: rupeshs <rupeshs@ami.com>
---
 arch/arm/dts/ast2705.dtsi | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/arch/arm/dts/ast2705.dtsi b/arch/arm/dts/ast2705.dtsi
index 2ada158cfc..7e5bd6e9f3 100644
--- a/arch/arm/dts/ast2705.dtsi
+++ b/arch/arm/dts/ast2705.dtsi
@@ -17,11 +17,8 @@
 				 <&soc1_rst SCU1_RESET_XPCS>;
 			reset-names = "stmmaceth", "ahb";
 			snps,axi-config = <&xgmac_axi_setup>;
-			snps,mtl-rx-config = <&xgmac_mtl_rx_setup>;
-			snps,mtl-tx-config = <&xgmac_mtl_tx_setup>;
 			snps,pbl = <8>;
 			phy-handle = <&xpcs0>;
-			pcs-handle = <&xpcs0>;
 			status = "disabled";
 
 			xgmac_axi_setup: stmmac-axi-config {
@@ -32,28 +29,9 @@
 				snps,blen = <0 0 0 0 0 8 4>;
 			};
 
-			xgmac_mtl_rx_setup: rx-queues-config {
-				snps,rx-queues-to-use = <1>;
-				snps,rx-sched-sp;
-
-				queue0 {
-					snps,dcb-algorithm;
-					snps,map-to-dma-channel = <0>;
-				};
-			};
-
-			xgmac_mtl_tx_setup: tx-queues-config {
-				snps,tx-queues-to-use = <1>;
-
-				queue0 {
-					snps,dcb-algorithm;
-				};
-			};
-
 			xgmac_mdio0: mdio {
 				#address-cells = <1>;
 				#size-cells = <0>;
-				compatible = "snps,dwmac-mdio";
 
 				xpcs0: ethernet-phy@0 {
 					reg = <0>;
-- 
2.43.0

